contracts/src/v0.8/functions/dev/1_0_0/RouterBase.sol

```diff
@@ -0,0 +1,370 @@
+  uint16 internal constant s_majorVersion = 1;
+  uint16 internal s_minorVersion = 0;
+  uint16 internal s_patchVersion = 0;
+
+  // ================================================================
+  // |                          Route state                         |
+  // ================================================================
+  mapping(bytes32 => address) internal s_route; /* id => contract address */
+  error RouteNotFound(bytes32 id);
+  // Use empty bytes to self-identify, since it does not have an id
+  bytes32 internal constant routerId = bytes32(0);
+
+  // ================================================================
+  // |                         Proposal state                       |
+  // ================================================================
+  uint8 internal constant MAX_PROPOSAL_SET_LENGTH = 8;
+
+  struct ProposalSet {
+    bytes32[] ids;
+    address[] from;
+    address[] to;
+    uint256 timelockEndBlock;
+  }
+  ProposalSet internal s_proposalSet;
+
+  event Proposed(
+    bytes32 proposalSetId,
+    address proposalSetFromAddress,
+    address proposalSetToAddress,
+    uint256 timelockEndBlock
+  );
+  event Upgraded(
+    bytes32 proposalSetId,
+    address proposalSetFromAddress,
+    address proposalSetToAddress,
+    uint16 major,
+    uint16 minor,
+    uint16 patch
+  );
+
+  struct ConfigProposal {
+    bytes32 fromHash;
+    bytes to;
+    uint256 timelockEndBlock;
+  }
+  mapping(bytes32 => ConfigProposal) internal s_proposedConfig; /* id => ConfigProposal */
+  event ConfigProposed(bytes32 id, bytes32 fromHash, bytes toBytes);
+  event ConfigUpdated(bytes32 id, bytes32 fromHash, bytes toBytes);
+  error InvalidProposal();
+  error ReservedLabel(bytes32 id);
+
+  // ================================================================
+  // |                          Timelock state                      |
+  // ================================================================
+  uint16 internal MAXIMUM_TIMELOCK_BLOCKS;
+  uint16 internal s_timelockBlocks;
+  struct TimeLockProposal {
+    uint16 from;
+    uint16 to;
+    uint256 timelockEndBlock;
+  }
+  TimeLockProposal s_timelockProposal;
+  event TimeLockProposed(uint16 from, uint16 to);
+  event TimeLockUpdated(uint16 from, uint16 to);
+  error ProposedTimelockAboveMaximum();
+  error TimelockInEffect();
+
+  // ================================================================
+  // |                          Config state                        |
+  // ================================================================
+  bytes32 internal s_config_hash;
+
+  // ================================================================
+  // |                       Initialization                         |
+  // ================================================================
+  constructor(
+    address newOwner,
+    uint16 timelockBlocks,
+    uint16 maximumTimelockBlocks,
+    bytes32[] memory initialIds,
+    address[] memory initialAddresses,
```

Comment:
Can we remove initialIds and initialAddresses to simplify? It might be even impossible to use them because the router needs to be deployed before the coordinator.